### DISCUSSION TITLE
add random() funciton that return a integer

### DISCUSSION THREAD

**Discussion by l1t1** [duckdb/duckdb#6142] [General]
As now I must do it by (random()*maxvalue)::int

**Comment by Mytherin** [comment]
Do you have a suggestion for what the syntax for that would be/what other systems have for the syntax?

**Comment by l1t1** [comment]
maybe random(low,high)?
sqlite 
ref https://www.sqlite.org/lang_corefunc.html#random
The random() function returns a pseudo-random integer between -9223372036854775808 and +9223372036854775807.
oracle
ref https://docs.oracle.com/cd/E11882_01/appdev.112/e40758/d_random.htm#ARPLS67507
The basic function gets a random number, greater than or equal to 0 and less than 1, with 38 digits to the right of the decimal (38-digit precision). Alternatively, you can get a random Oracle number x, where x is greater than or equal to low and less than high.

DBMS_RANDOM.VALUE(
  low  IN  NUMBER,
  high IN  NUMBER)
RETURN NUMBER;

python
Functions for integers
random.randrange(stop)random.randrange(start, stop[, step])
Return a randomly selected element from range(start, stop, step). This is equivalent to choice(range(start, stop, step)), but doesn’t actually build a range object.

The positional argument pattern matches that of [range()](mk:@MSITStore:D:\Python38\Doc\python388.chm::/library/stdtypes.html#range). Keyword arguments should not be used because the function may use them in unexpected ways.


Changed in version 3.2: [randrange()](mk:@MSITStore:D:\Python38\Doc\python388.chm::/library/random.html#random.randrange) is more sophisticated about producing equally distributed values. Formerly it used a style like int(random()*n) which could produce slightly uneven distributions.
random.randint(a, b)
Return a random integer N such that a <= N <= b. Alias for randrange(a, b+1).
